### DISCUSSION TITLE
I would like to define a different search path outside the connection string, It is a good Idea

### DISCUSSION THREAD

**Discussion by dnp1** [diesel-rs/diesel#4016] [Ideas]
Currently if I want my postgresql to use a different search path I have to include it in connection string.

What if I can choose a different schema in my `diesel.toml`?

The current schema field is only useful for schema.rs generation, but the tables and other stuff are all created on the public (or the default search_path to connection).

**Comment by weiznich** [comment]
This is currently not supported and would need much more support than just a field in your `diesel.toml`. 
In fact your `diesel.toml` file is solely used by diesel-cli to generate your schema.rs file.

> The current schema field is only useful for schema.rs generation, but the tables and other stuff are all created on the public (or the default search_path to connection).

I'm not sure that I understand this part. Tables and stuff is exactly created where you describe it in your migrations. If you don't specify a schema the default schema is used. If you specify a schema that schema is used.

**Reply by dnp1** [reply]
yes, but the diesel control table is always created in public (or the specified in the connection string) unless I put the schema inside connection string.

I can have a single database with different diesel "control" tables, why not?

**Reply by weiznich** [reply]
I'm sorry but I cannot follow what you are writing here. Can you provide a complete example of what you are trying to describe?

From what you've written so far I do see nothing that wouldn't already be possible with diesel today.
